src/com/owncloud/android/ui/activity/UploadFilesActivity.java

```diff
@@ -376,26 +439,39 @@ protected Boolean doInBackground(Void... params) {
          */
         @Override
         protected void onPostExecute(Boolean result) {
-            mCurrentDialog.dismiss();
-            mCurrentDialog = null;
+            if(requestCode == FileDisplayActivity.REQUEST_CODE__SELECT_FILES_FROM_FILE_SYSTEM) {
+                mCurrentDialog.dismiss();
+                mCurrentDialog = null;
+            }
             
             if (result) {
                 // return the list of selected files (success)
                 Intent data = new Intent();
-                data.putExtra(EXTRA_CHOSEN_FILES, mFileListFragment.getCheckedFilePaths());
+                if(requestCode == FileDisplayActivity.REQUEST_CODE__UPLOAD_FROM_CAMERA){
+                 data.putExtra(EXTRA_CHOSEN_FILES,new String[]{ capturedImagePath});
+                }
+                else {
+                    data.putExtra(EXTRA_CHOSEN_FILES, mFileListFragment.getCheckedFilePaths());
+                }
 
                 SharedPreferences.Editor appPreferencesEditor = PreferenceManager
                         .getDefaultSharedPreferences(getApplicationContext()).edit();
 
-
-                if (mRadioBtnMoveFiles.isChecked()){
-                    setResult(RESULT_OK_AND_MOVE, data);
-                    appPreferencesEditor.putInt("prefs_uploader_behaviour",
-                            FileUploader.LOCAL_BEHAVIOUR_MOVE);
-                } else {
+                if(requestCode == FileDisplayActivity.REQUEST_CODE__UPLOAD_FROM_CAMERA){
                     setResult(RESULT_OK, data);
```

Comment:
This result should be RESULT_OK_AND_MOVE, since you are moving the file, shouldn't it?